System/ContentFile.cpp

```diff
@@ -100,6 +106,93 @@ namespace RTE {
 		m_DataPathAndReaderPosition = m_DataPath + "\n" + newPosition;
 	}
 
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	int ContentFile::GetImageFileInfo(ImageFileInfoType infoTypeToGet) {
+		bool fileInfoIsEmpty = false;
+		for (const int &fileInfoEntry : m_ImageFileInfo) {
+			if (fileInfoEntry == -1) {
+				fileInfoIsEmpty = true;
+				break;
+			}
+		}
+		if (fileInfoIsEmpty) {
+			FILE *imageFile = fopen(m_DataPath.c_str(), "rb");
+			RTEAssert(imageFile, "Failed to open file prior to reading info of image file with following path and name:\n\n" + m_DataPath + "\n\nThe file may not exist or be corrupt.");
+
+			if (m_DataPathExtension == ".png") {
+				ReadAndStorePNGFileInfo(imageFile);
+			} else if (m_DataPathExtension == ".bmp") {
+				ReadAndStoreBMPFileInfo(imageFile);
+			} else {
+				RTEAbort("Somehow ended up attempting to read image file info for an unsupported image file type.\nThe file path and name were:\n\n" + m_DataPath);
+			}
+			fclose(imageFile);
+		}
+		return m_ImageFileInfo[infoTypeToGet];
+	}
+
+/////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+	void ContentFile::ReadAndStorePNGFileInfo(FILE *imageFile) {
+		std::array<uint8_t, 8> fileSignature = {};
+
+		// Read the first 8 bytes to then verify they match the PNG file signature. Which is { 137, 80, 78, 71, 13, 10, 26, 10 } or { '\211', 'P', 'N', 'G', '\r', '\n', '\032', '\n' }.
+		fread(fileSignature.data(), sizeof(uint8_t), fileSignature.size(), imageFile);
+		if (png_sig_cmp(fileSignature.data(), 0, fileSignature.size()) == 0) {
+			png_structp pngReadStruct = png_create_read_struct(PNG_LIBPNG_VER_STRING, nullptr, nullptr, nullptr);
+			png_infop pngInfo = png_create_info_struct(pngReadStruct);
+
+			png_init_io(pngReadStruct, imageFile);
+			// Set the PNG reader to skip the first 8 bytes since we already handled them.
+			png_set_sig_bytes(pngReadStruct, fileSignature.size());
```

Comment:
I'm curious, does it shit itself if you just tried to read it without checking and skipping the first 8 bytes, if it's not a png?

Reply:
If you wish to go through the scientific method of fucking around and finding out do it at your own free time lel

What is likely to happen is you getting wrong data.

Reply:
Nah, if you don't know the answer already I'm not gonna bother lol. Feel free to resolve.